administrator/components/com_associations/views/association/view.html.php

```diff
@@ -147,11 +147,28 @@ protected function addToolbar()
 		$input = JFactory::getApplication()->input;
 		$input->set('hidemainmenu', 1);
 
+		$user       = JFactory::getUser();
+		$userId     = $user->id;
+		$checkedOut = !($this->table->checked_out == 0 || $this->table->checked_out == $userId);
+		$component  = $input->get('acomponent', '', 'string');
+
 		JToolbarHelper::title(JText::_('COM_ASSOCIATIONS_HEADER_EDIT'), 'contract');
 
-		JToolbarHelper::apply('reference', 'COM_ASSOCIATIONS_SAVE_REFERENCE');
+		$canEdit = $user->authorise('core.edit', $component . $this->referenceId);
+		if (isset($this->table->created_by))
+		{
+			$canEditOwn = $user->authorise('core.edit.own', $component . $this->referenceId) && $this->table->created_by == $userId;
+		}
+
```

Comment:
i'm no expert in this part.. but look at all other componets and the awser shown be there